### ISSUE TITLE
RandomExceptionCircuitBreakerIT.testBreakerWithRandomExceptions failing on master

### ISSUE THREAD

**Issue by pgomulka** [elastic/elasticsearch#67410]
**Build scan**:
https://gradle-enterprise.elastic.co/s/w3hd5wgzwm5ks on master
or 
https://gradle-enterprise.elastic.co/s/sd55fxrthungy on a PR based on master

**Repro line**:
`./gradlew ':server:internalClusterTest' --tests "org.elasticsearch.indices.memory.breaker.RandomExceptionCircuitBreakerIT.testBreakerWithRandomExceptions" -Dtests.seed=E2BE619DA31A6010 -Dtests.security.manager=true -Dtests.locale=ga -Dtests.timezone=Africa/Monrovia -Druntime.java=11`

**Reproduces locally?**:
yes

**Applicable branches**:
master

**Failure history**:
https://build-stats.elastic.co/app/kibana#/discover?_g=(refreshInterval:(pause:!t,value:0),time:(from:now-7d,mode:quick,to:now))&_a=(columns:!(_source),index:b646ed00-7efc-11e8-bf69-63c8ef516157,interval:auto,query:(language:lucene,query:%22RandomExceptionCircuitBreakerIT%22),sort:!(process.time-start,desc))

looks like it is introduced on 12th Jan 2021
**Failure excerpt**:
```
Error Message
java.lang.AssertionError: timed out waiting for green state
Stacktrace
java.lang.AssertionError: timed out waiting for green state
	at __randomizedtesting.SeedInfo.seed([E2BE619DA31A6010:73E22FE1740B6DE3]:0)
	at org.junit.Assert.fail(Assert.java:88)
	at org.elasticsearch.test.ESIntegTestCase.ensureColor(ESIntegTestCase.java:912)
	at org.elasticsearch.test.ESIntegTestCase.ensureGreen(ESIntegTestCase.java:865)
	at org.elasticsearch.test.ESIntegTestCase.ensureGreen(ESIntegTestCase.java:854)
	at org.elasticsearch.indices.memory.breaker.RandomExceptionCircuitBreakerIT.testBreakerWithRandomExceptions(RandomExceptionCircuitBreakerIT.java:182)

```
but in logs this also looks suspicious
```
> Caused by: java.lang.IllegalStateException: requires a CodecReader or a SequentialStoredFieldsLeafReader, got class org.elasticsearch.test.engine.ThrowingLeafReaderWrapper
  1>    at org.elasticsearch.common.lucene.index.SequentialStoredFieldsLeafReader.getSequentialStoredFieldsReader(SequentialStoredFieldsLeafReader.java:62)
  1>    at org.elasticsearch.index.engine.LuceneChangesSnapshot.readDocAsOp(LuceneCha  1> ngesSnapshot.java:234)
  1>    at org.elasticsearch.index.engine.LuceneChangesSnapshot.next(LuceneChangesSnapshot.java:127)
  1>    at org.elasticsearch.indices.recovery.RecoverySourceHandler$OperationBatchSender.nextChunkRequest(RecoverySourceHandler.java:751)
  1>    at org.elasticsearch.indices.recovery.RecoverySourceHandler$OperationBatchSender.nextChunkRequest(RecoverySourceHandler.java:716)
  1>    at org.elasticsearch.indices.recovery.MultiChunkTransfer.getNextRequest(MultiChunkTransfer.java:162)
  1>    ... 34 more

```

**Comment by dnhatn** [comment]
This is fixed in 5fe0d67ade5fed3de5b062a1a0837fd02286e0a9.

**Comment by stu-elastic** [comment]
There was another instance of this:
**Build scan**: https://gradle-enterprise.elastic.co/s/apoi7fe4yi4wo/

**Repro line**: ```
./gradlew ':server:internalClusterTest' --tests "org.elasticsearch.indices.memory.breaker.RandomExceptionCircuitBreakerIT.testBreakerWithRandomExceptions" -Dtests.seed=74006D604F948D1D -Dtests.security.manager=true -Dtests.locale=sl-SI -Dtests.timezone=Africa/Tripoli -Druntime.java=11     
```

**Reproduces locally?**:
yes

**Applicable branches**:
master
